### PR TITLE
Feature LCS-179 - tenant details, request journey

### PR DESCRIPTION
**Pull request by joefitter**
* added fields to tenant details step
* added country typeahead
* updated loop to handle third case
* added acceptance tests

### CODE REVIEW

File: apps/end-tenancy/controllers/loop.js [hunk 1]
```diff
@@ -36,21 +36,60 @@ module.exports = class LoopController extends DateController {
     return Object.keys(this.options.subSteps).indexOf(step) > -1;
   }
 
+  getStep(req) {
+    return this.options.subSteps[req.params.action];
+  }
+
+  redirectTo(step, req, res) {
+    return res.redirect(`${req.baseUrl.replace(/\/$/, '')}${this.options.route.replace(/\/$/, '')}/${step}`);
```
Comment (JoeChapman) [inline apps/end-tenancy/controllers/loop.js]: Does this pass the line length linting?

File: apps/end-tenancy/views/partials/items-table.html [hunk 2]
```diff
@@ -17,7 +17,7 @@
           {{#fields}}
             <td>
               {{value}}
-              <a href="{{baseUrl}}/{{route}}/{{field}}/{{id}}/change" class="edit-link">{{editText}}</a>
+              <a href="{{baseUrl}}/{{route}}/{{subroute}}/{{id}}/change" class="edit-link">{{editText}}</a>
```
Comment (JoeChapman) [inline apps/end-tenancy/views/partials/items-table.html]: Will the locals always be defined?

Comment (joefitter) [inline apps/end-tenancy/views/partials/items-table.html]: in this case yeah, as this partial is only used with the loop controller, which defines each local

File: apps/end-tenancy/controllers/loop.js (hunk 1 above)
Comment (joefitter) [inline apps/end-tenancy/controllers/loop.js]: surprisingly yes, must be close though

**Pull request merged**